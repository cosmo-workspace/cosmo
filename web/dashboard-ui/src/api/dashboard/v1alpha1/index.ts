/* tslint:disable */
/* eslint-disable */
/**
 * Cosmo Dashboard API
 * Manipulate cosmo dashboard resource API
 *
 * The version of the OpenAPI document: v1alpha1
 * Contact: jlandowner8@gmail.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


export * from "./api";
export * from "./configuration";

