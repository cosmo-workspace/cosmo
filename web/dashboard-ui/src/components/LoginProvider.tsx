import { Box, CircularProgress, CssBaseline, Stack, Typography } from '@mui/material';
import { useSnackbar } from 'notistack';
import React, { createContext, useContext, useEffect, useState } from 'react';
import { User } from '../proto/gen/dashboard/v1alpha1/user_pb';
import { useAuthService, useUserService } from '../services/DashboardServices';
import { useProgress } from './ProgressProvider';


/**
 * context
 * ref: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24509#issuecomment-774430643
 */
const Context = createContext<ReturnType<typeof useLoginModule>>(undefined as any);

/**
 * hooks
 */
export function useLogin() {
  return useContext(Context);
}

const useLoginModule = () => {
  console.log('useLoginModule');
  const [loginUser, setLoginUser] = useState<User>();
  const { enqueueSnackbar } = useSnackbar();
  const { setMask, releaseMask } = useProgress();
  const userService = useUserService();
  const authService = useAuthService();


  /**
   * verifyLogin
   */
  const verifyLogin = async () => {
    console.log('verify start');
    try {
      const resp = await authService.verify({});
      if (resp.userName) {
        await getMyUserInfo(resp.userName);
      }
    }
    catch (error) {
      setLoginUser(undefined);
      //handleError(error);
    }
    finally {
      console.log('verify end');
    }
  }


  /**
   * login: SignIn 
   */
  const login = async (userName: string, password: string) => {
    console.log('login');
    try {
      const res = await authService.login({ userName: userName, password: password });
      await getMyUserInfo(userName);
      console.log('login end');
      return res;
    }
    catch (error) {
      setLoginUser(undefined);
      console.log('login error');
      handleError(error);
      throw error;
    }
  }


  /**
   * login: MyUserInfo
   */
  const getMyUserInfo = async (userName: string) => {
    console.log('getMyUserInfo', userName);
    // if (loginUser || !username) {
    //   console.log('getMyUserInfo cancel', loginUser, username);
    //   return;
    // }
    try {
      const responseUser = await userService.getUser({ userName: userName });
      setLoginUser(prev => { console.log('setLoginUser', prev, responseUser.user); return responseUser.user! });
    }
    catch (error) {
      console.log('getMyUserInfo error');
      handleError(error);
      throw error;
    }
  }

  /**
   * login: MyUserInfo
   */
  const refreshUserInfo = async () => {
    console.log('refreshUserInfo');
    if (loginUser) {
      getMyUserInfo(loginUser.name);
    }
  }

  /**
   * logout: Pagetemplate 
   */
  const logout = async () => {
    console.log('logout');
    try {
      await authService.logout({});
    }
    catch (error) {
      console.log('logout error');
      handleError(error);
      throw error;
    }
    finally {
      console.log('logout end');
      setLoginUser(prev => { console.log('setLoginUser', `${prev} -> undefined`); return undefined });
    }
  }

  /**
   * updataPassword
   */
  const updataPassword = async (currentPassword: string, newPassword: string) => {
    console.log('updataPassword', loginUser?.name);
    setMask();
    try {
      try {
        return await userService.updateUserPassword({ userName: loginUser!.name, currentPassword, newPassword });
      }
      catch (error) {
        handleError(error);
        throw error;
      }
    }
    finally {
      console.log('updataPassword end');
      releaseMask();
    }
  }

  /**
   * clearLoginUser 
   */
  const clearLoginUser = async () => {
    setLoginUser(undefined);
  }

  /**
   * error handler
   */
  const handleError = (error: any) => {
    console.log(error);
    const msg = error?.message;
    msg && enqueueSnackbar(msg, { variant: 'error' });
  }

  return {
    loginUser,
    verifyLogin,
    login,
    logout,
    refreshUserInfo,
    updataPassword,
    clearLoginUser,
  };
}

/**
 * Provider
 */
export const LoginProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
  console.log('LoginProvider');
  const loginModule = useLoginModule();
  const [isVerified, setIsVerified] = useState(false);

  useEffect(() => {
    loginModule.verifyLogin()
      .then(() => setIsVerified(true));
  }, []); // eslint-disable-line

  return (
    <Context.Provider value={loginModule}>
      {isVerified ? children :
        <div>
          <CssBaseline />
          <Stack sx={{ m: 10 }} alignItems='center' spacing={2}>
            <Box sx={{ display: 'flex' }}>
              <CircularProgress />
            </Box>
            <Typography >verifying session...</Typography>
          </Stack>
        </div>}
    </Context.Provider >
  )
}

/**
 * export private member. (for test)
 */
// export const __local__ = {
//   useLoginModule,
// };
