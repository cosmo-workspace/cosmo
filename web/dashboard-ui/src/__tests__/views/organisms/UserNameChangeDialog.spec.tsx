import { Button } from "@mui/material";
import '@testing-library/jest-dom';
import { act, cleanup, render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import { useSnackbar } from "notistack";
import React from "react";
import { MockedFunction, afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import { useLogin } from "../../../components/LoginProvider";
import { User } from "../../../proto/gen/dashboard/v1alpha1/user_pb";
import { useUserModule } from "../../../views/organisms/UserModule";
import { UserNameChangeDialog, UserNameChangeDialogContext } from "../../../views/organisms/UserNameChangeDialog";

//--------------------------------------------------
// mock definition
//--------------------------------------------------
vi.mock("notistack");
vi.mock("../../../components/LoginProvider");
vi.mock("../../../views/organisms/UserModule");

type MockedMemberFunction<T extends (...args: any) => any> = {
  [P in keyof ReturnType<T>]: MockedFunction<ReturnType<T>[P]>;
};

const useUserModuleMock = useUserModule as MockedFunction<typeof useUserModule>;
const userModuleMock: MockedMemberFunction<typeof useUserModule> = {
  existingRoles: [] as any,
  users: [] as any,
  getUsers: vi.fn(),
  createUser: vi.fn(),
  updateName: vi.fn(),
  updateRole: vi.fn(),
  deleteUser: vi.fn(),
}
const useLoginMock = useLogin as MockedFunction<typeof useLogin>;
const loginMock: MockedMemberFunction<typeof useLogin> = {
  loginUser: {} as any,
  verifyLogin: vi.fn(),
  login: vi.fn(),
  logout: vi.fn(),
  updataPassword: vi.fn(),
  refreshUserInfo: vi.fn(),
  clearLoginUser: vi.fn(),
};

const useSnackbarMock = useSnackbar as MockedFunction<typeof useSnackbar>;
const snackbarMock: MockedMemberFunction<typeof useSnackbar> = {
  enqueueSnackbar: vi.fn(),
  closeSnackbar: vi.fn(),
};

const closeHandlerMock = vi.fn();

//--------------------------------------------------
// test
//--------------------------------------------------
describe("UserNameChangeDialog", () => {

  beforeEach(async () => {
    useSnackbarMock.mockReturnValue(snackbarMock);
    useUserModuleMock.mockReturnValue(userModuleMock);
    useLoginMock.mockReturnValue(loginMock);
  });

  afterEach(() => {
    vi.restoreAllMocks();
    cleanup();
  });


  describe("render", () => {

    it("render", async () => {
      const user1: User = new User({ name: 'user1', roles: ["CosmoAdmin"], displayName: 'user1 name' });
      render(
        <UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />
      );
      expect(document.body).toMatchSnapshot();
    });

  });


  describe("behavior", () => {

    const user1: User = new User({ name: 'user1', roles: ["CosmoAdmin"], displayName: 'user1 name' });

    it("ok", async () => {
      const user = userEvent.setup();
      const { baseElement } = render(<UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />);

      const nameElement = baseElement.querySelector('[name="name"]')!;
      await user.type(nameElement, 'New Name', { initialSelectionStart: 0, initialSelectionEnd: 99 });

      userModuleMock.updateName.mockResolvedValue({} as any);
      loginMock.refreshUserInfo.mockResolvedValue({} as any);
      await user.click(screen.getByText('Update'));
      expect(userModuleMock.updateName.mock.calls).toMatchObject([["user1", "New Name"]]);
      expect(loginMock.refreshUserInfo.mock.calls).toMatchObject([[]]);
      expect(closeHandlerMock.mock.calls.length).toEqual(1);
    });

    it("ng required", async () => {
      const user = userEvent.setup();
      const { baseElement } = render(<UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />);

      expect(screen.getAllByText('User Name')[0].parentElement!
        .getElementsByClassName('MuiFormHelperText-root')[0]).toBeUndefined();

      const nameElement = baseElement.querySelector('[name="name"]')!;
      await user.type(nameElement, '{backspace}', { initialSelectionStart: 0, initialSelectionEnd: 99 });
      await act(async () => {
        await user.click(screen.getByText('Update'));
      });
      expect(screen.getAllByText('User Name')[0].parentElement!
        .getElementsByClassName('MuiFormHelperText-root')[0]).toHaveTextContent('Required');
      expect(userModuleMock.updateName.mock.calls.length).toEqual(0);
      expect(loginMock.refreshUserInfo.mock.calls.length).toEqual(0);
      expect(closeHandlerMock.mock.calls.length).toEqual(0);
    });

    it("ng over 32", async () => {
      const user = userEvent.setup();
      const { baseElement } = render(<UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />);

      expect(screen.getAllByText('User Name')[0].parentElement!
        .getElementsByClassName('MuiFormHelperText-root')[0]).toBeUndefined();

      const nameElement = baseElement.querySelector('[name="name"]')!;
      await user.type(nameElement, '----+----1----+----2----+----3--x', { initialSelectionStart: 0, initialSelectionEnd: 99 });
      await act(async () => {
        await user.click(screen.getByText('Update'));
      });

      expect(screen.getAllByText('User Name')[0].parentElement!
        .getElementsByClassName('MuiFormHelperText-root')[0]).toHaveTextContent('Max 32 characters');
      expect(userModuleMock.updateName.mock.calls.length).toEqual(0);
      expect(loginMock.refreshUserInfo.mock.calls.length).toEqual(0);
      expect(closeHandlerMock.mock.calls.length).toEqual(0);

      await act(async () => {
        await user.type(nameElement, '----+----1----+----2----+----3--', { initialSelectionStart: 0, initialSelectionEnd: 99 });
      });
      expect(screen.getAllByText('User Name')[0].parentElement!
        .getElementsByClassName('MuiFormHelperText-root')[0]).toBeUndefined();
    });

    it("cancel", async () => {
      const user = userEvent.setup();
      render(<UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />);
      await user.click(screen.getByText('Cancel'));
      expect(userModuleMock.updateName.mock.calls.length).toEqual(0);
      expect(loginMock.refreshUserInfo.mock.calls.length).toEqual(0);
      expect(closeHandlerMock.mock.calls.length).toEqual(1);
    });

    it("not close <- click outside", async () => {
      const user = userEvent.setup();
      const { baseElement } = render(<UserNameChangeDialog onClose={() => closeHandlerMock()} user={user1} />);
      await user.click(baseElement.getElementsByClassName('MuiDialog-container')[0]);
      expect(userModuleMock.updateName.mock.calls.length).toEqual(0);
      expect(loginMock.refreshUserInfo.mock.calls.length).toEqual(0);
      expect(closeHandlerMock.mock.calls.length).toEqual(0);
    });

  });


  describe("UserNameChangeDialogContext", () => {

    it("open/close", async () => {

      const Stub = () => {
        const dispatch = UserNameChangeDialogContext.useDispatch();
        const user1: User = new User({ name: 'user1', displayName: 'user1name' });
        return (<>
          <Button onClick={() => dispatch(true, { user: user1 })}>open</Button>
          <Button onClick={() => dispatch(false)}>close</Button>
        </>);
      }
      const user = userEvent.setup();
      render(<UserNameChangeDialogContext.Provider><Stub /></UserNameChangeDialogContext.Provider>);
      await act(async () => { expect(document.body).toMatchSnapshot('1.initial render'); });
      await act(async () => { await user.click(screen.getByText('open')); });
      await act(async () => { expect(document.body).toMatchSnapshot('2.opend'); });
      await act(async () => { await user.click(screen.getByText('close')); });
      await act(async () => { expect(document.body).toMatchSnapshot('3.closed'); });
    });

  });

});