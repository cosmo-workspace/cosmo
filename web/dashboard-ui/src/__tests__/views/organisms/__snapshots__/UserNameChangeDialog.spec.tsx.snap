// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNameChangeDialog UserNameChangeDialogContext open/close: 1.initial render 1`] = `
<body
  style=""
>
  <div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`UserNameChangeDialog UserNameChangeDialogContext open/close: 2.opend 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="mui-0000_for_toMatchSnapShot"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth css-twia2z-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="mui-0000_for_toMatchSnapShot"
        >
          Change user name
        </h2>
        <form>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="css-1lwbda4-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-0000_for_toMatchSnapShot"
                  id="mui-0000_for_toMatchSnapShot-label"
                >
                  User ID
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-5z923u-MuiInputBase-root-MuiFilledInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PersonOutlineTwoToneIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="8"
                        opacity=".3"
                        r="2.1"
                      />
                      <path
                        d="M12 14.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V17c0-.64-3.13-2.1-6.1-2.1z"
                        opacity=".3"
                      />
                      <path
                        d="M12 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm6.1 5.1H5.9V17c0-.64 3.13-2.1 6.1-2.1s6.1 1.46 6.1 2.1v1.1zM12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6.1c1.16 0 2.1.94 2.1 2.1 0 1.16-.94 2.1-2.1 2.1S9.9 9.16 9.9 8c0-1.16.94-2.1 2.1-2.1z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
                    id="mui-0000_for_toMatchSnapShot"
                    readonly=""
                    tabindex="-1"
                    type="text"
                    value="user1"
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-0000_for_toMatchSnapShot"
                  id="mui-0000_for_toMatchSnapShot-label"
                >
                  User Name
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PersonOutlineTwoToneIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="8"
                        opacity=".3"
                        r="2.1"
                      />
                      <path
                        d="M12 14.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V17c0-.64-3.13-2.1-6.1-2.1z"
                        opacity=".3"
                      />
                      <path
                        d="M12 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm6.1 5.1H5.9V17c0-.64 3.13-2.1 6.1-2.1s6.1 1.46 6.1 2.1v1.1zM12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6.1c1.16 0 2.1.94 2.1 2.1 0 1.16-.94 2.1-2.1 2.1S9.9 9.16 9.9 8c0-1.16.94-2.1 2.1-2.1z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-0000_for_toMatchSnapShot"
                    name="name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1in441m"
                    >
                      <span>
                        User Name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Update
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`UserNameChangeDialog UserNameChangeDialogContext open/close: 3.closed 1`] = `
<body
  style=""
>
  <div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      open
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      close
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`UserNameChangeDialog render render 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="mui-0000_for_toMatchSnapShot"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth css-twia2z-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id="mui-0000_for_toMatchSnapShot"
        >
          Change user name
        </h2>
        <form>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="css-1lwbda4-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-0000_for_toMatchSnapShot"
                  id="mui-0000_for_toMatchSnapShot-label"
                >
                  User ID
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-5z923u-MuiInputBase-root-MuiFilledInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PersonOutlineTwoToneIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="8"
                        opacity=".3"
                        r="2.1"
                      />
                      <path
                        d="M12 14.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V17c0-.64-3.13-2.1-6.1-2.1z"
                        opacity=".3"
                      />
                      <path
                        d="M12 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm6.1 5.1H5.9V17c0-.64 3.13-2.1 6.1-2.1s6.1 1.46 6.1 2.1v1.1zM12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6.1c1.16 0 2.1.94 2.1 2.1 0 1.16-.94 2.1-2.1 2.1S9.9 9.16 9.9 8c0-1.16.94-2.1 2.1-2.1z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
                    id="mui-0000_for_toMatchSnapShot"
                    readonly=""
                    tabindex="-1"
                    type="text"
                    value="user1"
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-0000_for_toMatchSnapShot"
                  id="mui-0000_for_toMatchSnapShot-label"
                >
                  User Name
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="PersonOutlineTwoToneIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="8"
                        opacity=".3"
                        r="2.1"
                      />
                      <path
                        d="M12 14.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V17c0-.64-3.13-2.1-6.1-2.1z"
                        opacity=".3"
                      />
                      <path
                        d="M12 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4zm6.1 5.1H5.9V17c0-.64 3.13-2.1 6.1-2.1s6.1 1.46 6.1 2.1v1.1zM12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6.1c1.16 0 2.1.94 2.1 2.1 0 1.16-.94 2.1-2.1 2.1S9.9 9.16 9.9 8c0-1.16.94-2.1 2.1-2.1z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-0000_for_toMatchSnapShot"
                    name="name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1in441m"
                    >
                      <span>
                        User Name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Update
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
